python/lsst/analysis/tools/analysisMetrics/photometricRepeatabilityMetrics.py

```diff
@@ -0,0 +1,106 @@
+
+
+class StellarPhotometricRepeatabilityMetric(AnalysisMetric):
+    """Compute photometric repeatability from multiple measurements of a set of
+    stars. First, a set of per-source quality criteria are applied. Second,
+    the individual source measurements are grouped together by object index
+    and per-group quantities are computed (e.g., a representative S/N for the
+    group based on the median of associated per-source measurements). Third,
+    additional per-group criteria are applied. Fourth, summary statistics are
+    computed for the filtered groups.
+    """
+
+    fluxType: str = "psfFlux"
+
+    def setDefaults(self):
+        super().setDefaults()
+
+        # Apply per-source selection criteria
+        self.prep.selectors.bandSelector = BandSelector()
+
+        # Compute per-group quantities
+        self.process.buildActions.perGroupSn = PerGroupStatistic()
+        self.process.buildActions.perGroupSn.buildAction = Sn(fluxType=f"{self.fluxType}")
+        self.process.buildActions.perGroupSn.func = "median"
+        self.process.buildActions.perGroupExtendedness = PerGroupStatistic()
+        self.process.buildActions.perGroupExtendedness.buildAction.vectorKey = "extendedness"
+        self.process.buildActions.perGroupExtendedness.func = "median"
+        self.process.buildActions.perGroupCount = PerGroupStatistic()
+        self.process.buildActions.perGroupCount.buildAction.vectorKey = f"{self.fluxType}"
+        self.process.buildActions.perGroupStdev = PerGroupStatistic()
+        self.process.buildActions.perGroupStdev.buildAction = MagColumnNanoJansky(
+            vectorKey=f"{self.fluxType}"
+        )
+        self.process.buildActions.perGroupStdev.func = "std"
+
+        # Filter on per-group quantities
+        self.process.filterActions.perGroupStdevFiltered = MultiCriteriaDownselectVector(
+            vectorKey="perGroupStdev"
+        )
+        self.process.filterActions.perGroupStdevFiltered.selectors.count = ThresholdSelector(
+            vectorKey="perGroupCount",
+            op="ge",
+            threshold=3,
+        )
+        self.process.filterActions.perGroupStdevFiltered.selectors.sn = ThresholdSelector(
+            vectorKey="perGroupSn",
+            op="ge",
+            threshold=200,
+        )
+        self.process.filterActions.perGroupStdevFiltered.selectors.extendedness = ThresholdSelector(
+            vectorKey="perGroupExtendedness",
+            op="le",
+            threshold=0.5,
+        )
+
+        # Compute summary statistics on filtered groups
+        self.process.calculateActions.photRepeatStdev = MedianAction(vectorKey="perGroupStdevFiltered")
+        self.process.calculateActions.photRepeatOutlier = FracThreshold(
+            vectorKey="perGroupStdevFiltered",
+            op="ge",
+            threshold=0.015,
+            percent=True,
+        )
+
+        self.produce.units = {  # type: ignore
+            "photRepeatStdev": "mag",
```

Comment:
It would make more sense for the units to be mmag by default?

Reply:
Indeed, the specifications are stated in mmag, so converting to use mmag